src/components/fields/MarkerColor.js

```diff
@@ -93,35 +96,72 @@ class UnconnectedMarkerColor extends Component {
         container.marker.color.includes(MULTI_VALUED));
 
     return (
-      <Field {...this.props} multiValued={multiValued} attr={attr}>
-        <RadioBlocks
-          options={options}
-          activeOption={type}
-          onOptionChange={this.setType}
-        />
-        {!type ? null : type === 'constant' ? (
-          <Color
-            suppressMultiValuedMessage
-            attr="marker.color"
-            updatePlot={this.setValue}
-            fullValue={value.constant}
+      <Fragment>
+        <Field {...this.props} multiValued={multiValued} attr={attr}>
+          <RadioBlocks
+            options={options}
+            activeOption={type}
+            onOptionChange={this.setType}
           />
-        ) : container.marker &&
-        container.marker.colorsrc === MULTI_VALUED ? null : (
+          {!type ? null : type === 'constant' ? (
+            <Color
+              suppressMultiValuedMessage
+              attr="marker.color"
+              updatePlot={this.setValue}
+              fullValue={value.constant}
+            />
+          ) : container.marker &&
+          container.marker.colorsrc === MULTI_VALUED ? null : (
+            <Fragment>
+              <DataSelector suppressMultiValuedMessage attr="marker.color" />
+              {container.marker &&
+              container.marker.colorscale === MULTI_VALUED ? null : (
+                <Colorscale
+                  suppressMultiValuedMessage
+                  attr="marker.colorscale"
+                  updatePlot={this.setColorScale}
+                  colorscale={colorscale}
+                />
+              )}
+            </Fragment>
+          )}
+        </Field>
+        {type === 'constant' ? (
+          ''
+        ) : (
           <Fragment>
-            <DataSelector suppressMultiValuedMessage attr="marker.color" />
-            {container.marker &&
-            container.marker.colorscale === MULTI_VALUED ? null : (
-              <Colorscale
-                suppressMultiValuedMessage
-                attr="marker.colorscale"
-                updatePlot={this.setColorScale}
-                colorscale={colorscale}
-              />
-            )}
+            <Radio
```

Comment:
from the Percy screenshot it looks like multi-value is odd with these...

Reply:
well, I could be convinced that the percy screenshot is reasonable, let's talk about it